src/api/search/src/search.js

```diff
@@ -76,4 +76,81 @@ const search = async (
   };
 };
 
-module.exports = search;
+/**
+ * Advanced search allows you to look up multiple or single fields based on the input provided
+ * @param postText    - text to search in post field
+ * @param authorText  - text to search in author field
+ * @param titleText   - text to search in title field
+ * @param dateStart   - published after this date
+ * @param dateEnd     - published before this date
+ * @return all the results matching the fields text
+ * Range queries: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_ranges
+ * Match field queries: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query.html#query-dsl-match-query-zero
+ */
+const advancedSearch = async (
+  postText = '',
```

Comment:
Question: would it make more sense to pass in an Object with these options on it?  This would allow us to omit some of them.  Then, you could build your query Object by inspecting the values of the options passed in:

```js
const search = async (options) => {
  const query = {
    bool: {
      must: [],
    },
  };
  const must = query.bool.must;

  if(options.author) {
    must.push({
      match: {...},
    })
  }
  ...
  return query;
}
```

Reply:
The `zero_terms_query: 'all',` will omit if the value is not provided. 

The docs are explained here: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query.html#query-dsl-match-query-zero

Reply:
I don't think it makes sense to send values that we don't actually want, and to set defaults for things that were not passed to us by the query (e.g., setting things to `''`).  If a value isn't defined, don't add it.

Reply:
I think I should read about how the zero_terms_query works... but eventually this will ignore the field completely in the search... both ways will work. If its a matter of code looking clean I can change this. I'm not sure if the must.push will work though, in theory it looks logical but I will try.